src/Facade/Services/PartsStorageTypeFacadeService.cs

```diff
@@ -0,0 +1,122 @@
+
+        private readonly IRepository<StorageType, string> storageTypeRepository;
+
+        private readonly IRepository<PartsStorageType, PartsStorageTypeKey> partStorageTypeRepository;
+
+        public PartsStorageTypeFacadeService(
+            IRepository<PartsStorageType, PartsStorageTypeKey> partStorageTypeRepository,
+            ITransactionManager transactionManager,
+            IBuilder<PartsStorageType> resourceBuilder,
+            IRepository<Part, int> partRepository,
+            IRepository<StorageType, string> storageTypeRepository)
+            : base(partStorageTypeRepository, transactionManager, resourceBuilder)
+        {
+            this.partRepository = partRepository;
+            this.storageTypeRepository = storageTypeRepository;
+            this.partStorageTypeRepository = partStorageTypeRepository;
+        }
+
+        protected override async Task<PartsStorageType> CreateFromResourceAsync(
+            PartsStorageTypeResource resource,
+            IEnumerable<string> privileges = null)
+        {
+            var part = await this.partRepository.FindByAsync(p => p.PartNumber == resource.PartNumber);
+
+            if (part == null || string.IsNullOrEmpty(resource.PartNumber))
+            {
+                throw new PartsStorageTypeException("Part Number is empty or doesn't exist!");
+            }
+
+            var storageType = await this.storageTypeRepository.FindByAsync(st => st.StorageTypeCode == resource.StorageTypeCode);
+
+            if (storageType == null || string.IsNullOrEmpty(resource.StorageTypeCode))
+            {
+                throw new PartsStorageTypeException("Storage Type is empty or doesn't exist!");
+            }
+
+            var partStorageTypeAlreadyExists = await this.partStorageTypeRepository.FindByAsync(
+                                                                pst => pst.StorageTypeCode == resource.StorageTypeCode
+                                                                       && pst.PartNumber == resource.PartNumber);
+
+            if (partStorageTypeAlreadyExists != null)
+            {
+                throw new PartsStorageTypeException("Part Storage Type Already Exists");
+            }
+
+            return new PartsStorageType(
+                part,
+                storageType,
+                resource.Remarks,
+                resource.Maximum,
+                resource.Incr,
+                resource.Preference,
+                resource.BridgeId);
+        }
+
+        protected override async Task UpdateFromResourceAsync(
+            PartsStorageType entity,
+            PartsStorageTypeResource updateResource,
+            IEnumerable<string> privileges = null)
+        {
+            entity.Update(
+                updateResource.Remarks,
+                updateResource.Maximum,
+                updateResource.Incr,
+                updateResource.Preference,
+                updateResource.BridgeId);
+        }
```

Comment:
no awaits in here so you should have just chosen the normal UpdateFromResource to override